src/libdrakvuf/vmi.c

```diff
@@ -300,6 +300,29 @@ event_response_t pre_mem_cb(vmi_instance_t vmi, vmi_event_t *event) {
 
         loop = loop->next;
     }
+
+    /* We need to call breakpoint handlers registered for this physical address */
+    struct wrapper *sbp = g_hash_table_lookup(drakvuf->breakpoint_lookup_pa, &s->memaccess.pa);
+    if (event->mem_event.out_access & VMI_MEMACCESS_X && sbp) {
```

Comment:
Probably better to do the hashtable lookup only after the check for VMI_MEMACCESS_X for performance reasons.